Xamarin.Forms.Platform.Android/AppCompat/Platform.cs

```diff
@@ -220,10 +220,25 @@ internal void SetPage(Page newRoot)
 			var layout = false;
 			if (Page != null)
 			{
-				_renderer.RemoveAllViews();
+				var removeList = new List<global::Android.Views.View>();
+				for (int i = 0; i < _renderer.ChildCount; i++)
+				{
+					removeList.Add(_renderer.GetChildAt(i));
+				}
+
+				var renderersToDispose = _navModel.Roots.Select(Android.Platform.GetRenderer).ToList();
+
+				Device.StartTimer(TimeSpan.FromSeconds(1), () =>
```

Comment:
It also wasn't meant to be our permanent solution. It was intentionally exaggerated and submitted for our UI tests to run through to see what sort of side effects it might cause. 😄